tiledb/query.py

```diff
@@ -14,20 +19,272 @@ def __init__(
         self,
         array: Array,
         ctx: Ctx = None,
+        attrs=None,
+        cond=None,
+        dims=None,
+        coords=False,
+        index_col=True,
+        order=None,
+        use_arrow=None,
+        return_arrow=False,
+        return_incomplete=False,
     ):
-        """Class representing a query on a TileDB Array.
 
-        :param array: tiledb.Array the query is on
-        :param ctx: A TileDB Context
-        """
+        """Class representing a query on a TileDB Array."""
+
+        if array.mode not in ("r", "d"):
+            raise ValueError("array mode must be read or delete mode")
+
+        if dims is not None and coords == True:
+            raise ValueError("Cannot pass both dims and coords=True to Query")
+
+        if return_incomplete and not array.schema.sparse:
+            raise TileDBError(
+                "Incomplete queries are only supported for sparse arrays at this time"
+            )
+
+        # reference to the array we are querying
         self._array = array
         super().__init__(
             ctx, lt.Array(ctx if ctx is not None else default_ctx(), array)
         )
 
+        dims_to_set = list()
+        self._dims = None
+
+        if dims is False:
+            self._dims = False
+        elif dims != None and dims != True:
+            domain = array.schema.domain
+            for dname in dims:
+                if not domain.has_dim(dname):
+                    raise TileDBError(f"Selected dimension does not exist: '{dname}'")
+            self._dims = dims
+        elif coords == True or dims == True:
+            domain = array.schema.domain
+            self._dims = [domain.dim(i).name for i in range(domain.ndim)]
```

Comment:
I know this code is pre-existed and you just copied it over, but I really want to see this simplied. I think 

> `dims != None and dims != True` == 
> `dims != None and dims == False` == 
> `dims == False` 

which is already accounted for in the first conditional block. 

```suggestion
        if dims == False:
            self._dims = False
        elif coords or dims: 
            domain = array.schema.domain
            self._dims = [domain.dim(i).name for i in range(domain.ndim)]
```

Reply:
Oh wait I'm confused. `dims` is `Optional[bool | Sequence[str]`?

Reply:
yes

Reply:
I don't think you code simplification is possible.

Reply:
I added my own one. Checking that dimension is present seems like something we need.